src/scripting/lua_widget_attributes.cpp

```diff
@@ -342,16 +428,316 @@ WIDGET_SETTER("marked_up_text", t_string, gui2::styled_widget)
 	w.set_label(value);
 }
 
+WIDGET_GETTER("characters_per_line", int, gui2::label)
+{
+	return w.get_characters_per_line();
+}
+
+WIDGET_SETTER("characters_per_line", int, gui2::label)
+{
+	if(value < 0) {
+		throw std::invalid_argument("characters_per_line must be >= 0");
+	}
+	w.set_characters_per_line(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("editable", bool, gui2::text_box)
+{
+	return w.is_editable();
+}
+
+WIDGET_SETTER("editable", bool, gui2::text_box)
+{
+	w.set_editable(value);
+}
+
+WIDGET_GETTER("ellipsize_mode", std::string, gui2::styled_widget)
+{
+	std::string s;
+
+	switch(w.get_text_ellipse_mode()) {
+		case(PangoEllipsizeMode::PANGO_ELLIPSIZE_NONE):
+			s = "none";
```

Comment:
could have done just `return "none"`, same for the cases later. `s` and the break statements were not needed.

Reply:
could have also added a `decode_ellipsization_mode` in helper.cpp and done just `return decode_ellipsization_mode(w.get_text_ellipse_mode())`.

Reply:
Simplification done in 2e0f8c453e2caace51cd30804c6093727af94fda.